### PR TITLE
Attempt to enable merge queue

### PR DESCRIPTION
**Pull request by jamilbk**
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#merge_group

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 name: Continuous Delivery
 on:
+  merge_group:
```
Comment (AndrewDryga) [inline .github/workflows/cd.yml]: I think we don't need to have it here since this one only runs after stuff is merged

Comment (jamilbk) [reply .github/workflows/cd.yml]: Good catch, fixed.

**Review by thomaseizinger — APPROVED** [review]
I think `types` is implicit, up to you!

**Comment by thomaseizinger** [comment]
Something seems off with the smoke-test, I'll investigate.

**Comment by jamilbk** [comment]
> I think `types` is implicit, up to you!

Yeah just figured it'll prevent surprises later if the behavior changes due to beta status.

**Comment by jamilbk** [comment]
@AndrewDryga Build seems to be failing due to an intermittent container build issue -- maybe a cache issue? I noticed we're over our cache limit. I can look into increasing that if needed.

The `type_check` workflow is failing due to an Elixir issue: https://github.com/elixir-lang/elixir/issues/12691

**Comment by jamilbk** [comment]
@AndrewDryga Going to merge this so that these settings take effect for the remaining PRs. Currently, merge queue won't wait for all status checks to pass before merging the PR.

**Pull request merged**